cli/src/config.rs

```diff
@@ -434,6 +435,41 @@ fn read_config_path(config_path: &Path) -> Result<config::Config, config::Config
         .build()
 }
 
+#[cfg(target_os = "macos")]
+fn config_dir() -> Option<PathBuf> {
+    // Define a closure to check if config file exists in a directory
+    let check_config = |dir: &Path| {
+        let mut config_file = dir.to_owned();
+        config_file.push("jj/config.toml");
+        if config_file.exists() {
+            Some(dir.to_owned())
+        } else {
+            None
+        }
+    };
+
+    // Try getting config directory from Apple base strategy. `Application Support`
+    // is really the data_dir, so get that.
+    let path = etcetera::base_strategy::Apple::new()
+        .ok()
+        .and_then(|strategy| check_config(&strategy.data_dir()));
+
+    // If config directory is not found from Apple base strategy, try XDG instead
+    path.or_else(|| {
+        etcetera::choose_base_strategy()
+            .ok()
+            .and_then(|strategy| check_config(&strategy.config_dir()))
+    })
```

Comment:
I think we should prefer XDG's strategy over Apple so new users get this behavior by default when running `jj config set` or `jj config edit` without having an existing config in `~/Library/Application Supprt/`